scripts/generate-bootstraps.sh

```diff
@@ -382,23 +421,38 @@ while (($# > 0)); do
 done
 
 if [[ "$TERMUX_PACKAGE_MANAGER" == *" "* ]] || [[ " ${TERMUX_PACKAGE_MANAGERS[*]} " != *" $TERMUX_PACKAGE_MANAGER "* ]]; then
-	echo "[!] Invalid package manager '$TERMUX_PACKAGE_MANAGER'" 1>&2
-	echo "Supported package managers: '${TERMUX_PACKAGE_MANAGERS[*]}'" 1>&2
+	echo "[!] Invalid package manager '$TERMUX_PACKAGE_MANAGER'" >&2
+	echo "Supported package managers: '${TERMUX_PACKAGE_MANAGERS[*]}'" >&2
 	exit 1
 fi
 
 if [ -z "$REPO_BASE_URL" ]; then
-	echo "[!] The repository base url is not set." 1>&2
+	echo "[!] The repository base url is not set." >&2
 	exit 1
 fi
 
+if [ "$TERMUX_PACKAGE_DOWNLOAD_CACHE" = true ]; then
+	if [ "$TERMUX_PACKAGE_MANAGER" != apt ]; then
+		echo "[!] Option '--cache' does not support '$TERMUX_PACKAGE_MANAGER' yet." >&2
+		exit 1
+	fi
+	echo "[*] CACHE: Caching enabled. Cached packages will be stored and reused."
+	echo "[*] CACHE: Cache directory     = $TERMUX_PACKAGE_DOWNLOAD_CACHE_DIR"
+	echo "[*] CACHE: Temporary directory = $BOOTSTRAP_TMPDIR"
+	mkdir -p "$TERMUX_PACKAGE_DOWNLOAD_CACHE_DIR"
+	if [ ! -d "$TERMUX_PACKAGE_DOWNLOAD_CACHE_DIR" ]; then
+		echo "[!] CACHE: Failed to create cache directory '$TERMUX_PACKAGE_DOWNLOAD_CACHE_DIR'." >&2
+		exit 1
+	fi
+fi
+
 for package_arch in "${TERMUX_ARCHITECTURES[@]}"; do
 	BOOTSTRAP_ROOTFS="$BOOTSTRAP_TMPDIR/rootfs-${package_arch}"
 	BOOTSTRAP_PKGDIR="$BOOTSTRAP_TMPDIR/packages-${package_arch}"
 
 	# Create initial directories for $TERMUX_PREFIX
 	if ! ${BOOTSTRAP_ANDROID10_COMPATIBLE}; then
-		if [ ${TERMUX_PACKAGE_MANAGER} = "apt" ]; then
+		if [ "$TERMUX_PACKAGE_MANAGER" = apt ]; then
```

Comment:
Don't remove double quotes. `apt` is a string and not a keyword. It's convention to always use double quotes around variables and strings.

https://google.github.io/styleguide/shellguide.html#testing-strings